### PR TITLE
Run qFlex natively on Mac OSX using clang

### PR DESCRIPTION
**Pull request by s-mandra**

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -47,6 +47,25 @@ The following options for `configure` are also available:
...
+brew install coreutils gsl autoconf automake libomp pybind11
+```
+and through `pip3`:
+```
+pip3 install docopt pybind11 pytest cirq
+```
+After the installation of the required libraries, qFlex can be compiled on Mac
+OS X as:
+```
+autoreconf -i && autoconf && \
```
[the remark this answers is not in the document]
Comment (s-mandra) [reply docs/install.md]: Good point, I can try it.

**Comment by alankao64** [comment]
Can confirm that everything is working!

**Comment by s-mandra** [comment]
Let me try to automatize the process on Mac OS X. Otherwise, I'll merge this PR. Thanks!

File: docs/install.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (s-mandra) [reply docs/install.md]: Fixed. Now it realizes it's a Mac OS X and it runs few extra tests and set the right flags for the compilation.

**Comment by s-mandra** [comment]
> Can confirm that everything is working!

@alankao64, can you double check that everything is working by just running:
```
autoreconf -i && autoconf && ./configure && make -j8 && make -j8 run-tests
```
Thanks!

File: docs/install.md [hunk 2]
```diff
@@ -47,22 +47,17 @@ The following options for `configure` are also available:
...
 ```
 and through `pip3`:
 ```
-pip3 install pybind11 pytest cirq
-```
-After the installation of the required libraries, qFlex can be compiled on Mac
-OSX systems as:
-```
-autoreconf -i && autoconf && CXX=clang++ CPPFLAGS="-I$(brew --prefix)/include/ -L$(brew --prefix)/lib/" ./configure && make -j8 && make -j8 run-tests
+pip3 install docopt pybind11 pytest cirq
```
Comment (95-martin-orion) [inline docs/install.md]: It's probably worthwhile to keep the 

> Check prerequisites by running `autoreconf -i && autoconf && ./configure`

 line here, just to make it clear to users that autoconf is still required/usable on Mac.

File: configure.ac [hunk 3]
```diff
@@ -24,6 +24,20 @@ AC_ARG_ENABLE([all_checks],
    esac], [all_checks=true]
 )
 
+# Skip python checks
```
Comment (95-martin-orion) [inline configure.ac]: Fix comment (e.g. "Check for additional Mac OSX dependencies")

File: docs/install.md (hunk 2 above)
Comment (alexandrupaler) [reply docs/install.md]: The Python packages necessary for running/compiling QFlex are in requirements.txt

In the scripts folder the construct_cirq_environment.sh script calls the line
```python3 -m pip install -r scripts/requirements.txt```

**Comment by alankao64** [comment]
> @alankao64, can you double check that everything is working by just running:

Confirmed everything works after updating cirq!

File: docs/install.md (hunk 2 above)
Comment (s-mandra) [reply docs/install.md]: Thanks! Let me update the doc.

Comment (s-mandra) [reply docs/install.md]: Fixed.

File: docs/install.md [hunk 4]
```diff
@@ -53,11 +53,15 @@ qFlex can be natively compiled on Mac OS X by installing few dependencies throug
 `brew` and `pip3`. More precisely, the following libraries must be installed
 using `brew`:
 ```
-brew install coreutils gsl autoconf automake libomp pybind11
+$ brew install coreutils gsl autoconf automake libomp pybind11
 ```
 and through `pip3`:
 ```
-pip3 install docopt pybind11 pytest cirq
+# python3 -m pip install -r scripts/requirements.txt
```
Comment (95-martin-orion) [inline docs/install.md]: Probably want `$` instead of `#` here.

**Pull request merged**